### DISCUSSION TITLE
Deepness with SAM

### DISCUSSION THREAD

**Discussion by annaOrbica** [PUTvision/qgis-plugin-deepness#87] [Ideas]
Hey Deepness-Team,

first, really great plugin, congrats. I was playing around with Facebook's SAM recently and tried out the exported ONNX model with Deepness. It seemed like SAM was not compatible with the plugin (or I just missed something). Do you have any plans to adjust the plugin to be able to use SAM? 
Thanks in advance.

**Comment by bartoszptak** [comment]
Hi @annaOrbica!
A very insightful idea to use the SAM model in QGIS! Now we can only confirm that the SAM model is not compatible with the plugin. But we will put our attention to this interesting idea and will evaluate its possibilities of integration with the plugin. 

What is more interesting for you?
- automatic map segmentation based only on embeddings threshold
- automatic object on map segmentation based on positive/negative examples (e.g. by selecting mask layers as input)
- both above
